src/Styleguide/Elements/icons/CreditCardIcon.tsx

```diff
@@ -0,0 +1,148 @@
+import React from "react"
+
+export type CreditCardType =
+  | "mastercard"
+  | "visa"
+  | "discover"
+  | "amex"
+  | "other"
+
+export const CreditCardIcon = ({
+  type,
+  ...others
+}: React.SVGProps<SVGSVGElement> & {
+  type?: CreditCardType
+}) => {
+  switch (type) {
+    case "mastercard":
+      return <MastercardIcon {...others} />
+    case "visa":
+      return <VisaIcon {...others} />
+    case "discover":
+      return <DiscoverIcon {...others} />
+    case "amex":
+      return <AmexIcon {...others} />
+    case "other":
+    default:
+      return <FallbackIcon {...others} />
+  }
+}
+
+const AmexIcon = (props: React.SVGProps<SVGSVGElement>) => (
+  <svg
+    xmlnsXlink="http://www.w3.org/1999/xlink"
+    width={30}
+    height={20}
+    viewBox="0 0 30 20"
+    {...props}
+  >
+    <defs>
+      <rect id="a" width={30} height={20} rx={2} />
+    </defs>
+    <g fill="none" fillRule="evenodd">
+      <use fill="#EFEFEF" xlinkHref="#a" />
+      <rect width={29} height={19} x={0.5} y={0.5} stroke="#E5E5E5" rx={2} />
+      <g fill="#4DB3FF">
+        <path d="M9.999 11A.999.999 0 0 0 9 12.01v1.98A1 1 0 0 0 9.999 15H30v-4H9.999zM0 6v4h20.001A.999.999 0 0 0 21 8.99V7.01A.999.999 0 0 0 20.001 6H0z" />
+      </g>
+    </g>
+  </svg>
+)
+
+const DiscoverIcon = (props: React.SVGProps<SVGSVGElement>) => (
```

Comment:
I would have expected these to live in separate files but it's actually kind of nice/simple to list them all here.